### PR TITLE
generate weekView with js

### PR DESCRIPTION
**Pull request by andriussvilys**
- generate WeekView with js
- implemented controls in header (Today, next/prev week)
- clicking on month day now switches weekView
- date manipulation functions have been put in a separate module
- month/weekView are made into modules for use in header controls
- style update (button_today, selectedDate)

### CODE REVIEW

File: css/main.css [hunk 1]
```diff
@@ -5,6 +5,8 @@
     --button-bg: rgb(255, 255, 255);
     --primary: rgb(26,115,232);
     --primary-dark: rgb(25,103,210);
+    --secondary-variant: rgb(210,227,252);
+    --on-secondary-variant: rgb(24,90,188);
```
Comment (dovilel) [inline css/main.css:9]: what does "on" stand for?
Also, "secondary" is enough, no need for "variant"

File: css/main.css [hunk 2]
```diff
@@ -73,16 +79,18 @@ header{
...
 .button-controls{
     width: 2em;
+    opacity: 0.6;
 }
 .button-controls img{
     width: 100%;
 }
 
-.button-round{
+.button_round{
```
Comment (dovilel) [inline css/main.css:93]: why underscore?

File: js/dateManipulation.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+export const WEEKDAYS = 7
+
+export let selectedDate = new Date( Date.now() )
+
+export const isSameWeek = (date1, date2) => {
+    const week = getWeekDates(date1)
+    const result =  week.filter(item => isSameDate(item, date2))
```
Comment (dovilel) [inline js/dateManipulation.js:7]: Boolean(week.find(....))
would be a liiiitle bit faster and more succinct. You don't expect to filter more than one result anyway

File: js/dateManipulation.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export const getWeekDates = (date) => {
+
+    if(!date){
+        date = new Date(Date.now())
+    }
+
+    const weekDay = date.getDay()
+    const diffToMonday = (weekDay-1+7)%7
```
Comment (dovilel) [inline js/dateManipulation.js:29]: the 7 is back

File: js/header.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { switchMonth, findElemByDate, toggleSelectedDate } from "../month/months.js"
+import {switchWeekView} from "./weekView.js"
+import { selectedDate, setSelectedDate } from "./dateManipulation.js"
+
+const button_today = document.querySelector("#button_today")
+const headerControls_prev = document.querySelector("#headerControls_prev")
+const headerControls_next = document.querySelector("#headerControls_next")
+
+button_today.addEventListener('click', () => {
+    const today = new Date(Date.now())
```
Comment (dovilel) [inline js/header.js:10]: it's a tiny function, but you can probably create a function that returns today. You are already using it in a few places

File: js/header.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    handleNextPrevClick(-1)
+
+})
+
+headerControls_next.addEventListener('click', () => {
+    handleNextPrevClick(1)
+})
+
+const handleNextPrevClick = (direction) => {
+    const newDate = new Date( selectedDate.getFullYear(), selectedDate.getMonth(), selectedDate.getDate() + 7*direction )
```
Comment (dovilel) [inline js/header.js:26]: the seveeeen

Comment (dovilel) [reply js/header.js:26]: generally, would be nice idea to create a function that returns incremented day in your date stuff file

File: month/months.js [hunk 7]
```diff
@@ -1,70 +1,93 @@
...
-            class="button button-round monthView-button"
-            data-currentmonth=${content.currentMonth}
-            data-currentday=${content.currentDay}
-            >
-            ${content.date}
-        </button>
-    </div>
-    `
-    return element
+export function findElemByDate( date ){
```
Comment (dovilel) [inline month/months.js:38]: what Elem?

File: js/weekView.js [hunk 8]
```diff
@@ -1,9 +1,159 @@
-"use strict"
+import { isSameDate, isSameWeek, getWeekDates, selectedDate } from "../js/dateManipulation.js"
 
-const calendarDays = document.querySelector('.weekView-main')
+const ROW_COUNT = 25
```
Comment (dovilel) [inline js/weekView.js:3]: I strongly suspect your row count is soooomehow related to hours. Like 24 hours in the day. And you will definitely need that constant later.
Also, the only place you use ROW_COUNT has - 1 on it. So maybe it's 24 afterall?

File: js/weekView.js [hunk 9]
```diff
@@ -1,9 +1,159 @@
...
+
+    if(!isSameWeek(selectedDate, newDate)){
+
+        const wrapper = document.querySelector('.weekView-wrapper')
+        const weekView_current = document.querySelector('.weekView-main')
+        const weekView_new = generateWeekView(newDate)
+        
+        wrapper.appendChild(weekView_new)
+
+        weekView_current.classList.remove("slideIn")
```
Comment (dovilel) [inline js/weekView.js:145]: fancy :D 
a little nitpickiness - Google calendar slides it from different sides, depending on whether you go backwards or forwards

File: month/months.js [hunk 10]
```diff
@@ -1,70 +1,93 @@
...
+import { switchWeekView } from "../js/weekView.js"
+
 
 const CALENDAR_ROWS = 6
-const DAYS_IN_WEEK = 7
 
 const today = new Date(Date.now())
-let currentMonth = today.getMonth()
-let currentYear = today.getFullYear()
+let selectedDate_month = selectedDate
```
Comment (dovilel) [inline month/months.js:8]: something is wrong with variable naming. I'm not sure what it's supposed to be.
selectedMonth maybe?

File: month/months.js [hunk 11]
```diff
@@ -1,70 +1,93 @@
...
 
 document.querySelector("#months-next").addEventListener('click', () => {
-    incrementMonth(1)
-    fillMonth(currentYear, currentMonth)
+    const nextDate = new Date(selectedDate_month.getFullYear(), selectedDate_month.getMonth() + 1, 1)
```
Comment (dovilel) [inline month/months.js:11]: function to increment month would be nice

Comment (dovilel) [reply month/months.js:11]: I mean, you had it even. Why delete?

File: month/months.js [hunk 12]
```diff
@@ -1,70 +1,93 @@
...
-        currentMonth = currentMonth%12
-        currentYear++
-    }
-    if(currentMonth < 0){
-        currentMonth += 12
-        currentYear--
-    }
+export function toggleSelectedDate(element){
+    const buttons = document.querySelectorAll('#month button')
+    buttons.forEach(elem => {
```
Comment (dovilel) [inline month/months.js:32]: can you select by class, instead of iterating through all buttons?

File: month/months.js [hunk 13]
```diff
@@ -1,70 +1,93 @@
...
-    currentMonth += value
-    if(currentMonth > 11){
-        currentMonth = currentMonth%12
-        currentYear++
-    }
-    if(currentMonth < 0){
-        currentMonth += 12
-        currentYear--
-    }
+export function toggleSelectedDate(element){
```
Comment (dovilel) [inline month/months.js:30]: element - need better naming

File: month/months.js [hunk 14]
```diff
@@ -1,70 +1,93 @@
...
-            data-currentday=${content.currentDay}
-            >
-            ${content.date}
-        </button>
-    </div>
-    `
-    return element
+export function findElemByDate( date ){
+
+    const buttons = Array.from( document.querySelectorAll("#month button") )
```
Comment (dovilel) [inline month/months.js:40]: considering you already gave data attribute on the element, you can just select by it

File: month/months.js [hunk 15]
```diff
@@ -1,70 +1,93 @@
...
     const months = document.querySelector("#month")
-    months.innerHTML = html
+    months.innerHTML = ''
+    const monthView = getMonthViewDays(newDate);
+
+    monthView.forEach(date => {
+        months.appendChild( generateDayCell(date) )
+    })
+
+    // setSelectedDate(newDate)
```
Comment (dovilel) [inline month/months.js:82]: try not to leave commented out code

File: css/main.css (hunk 1 above)
Comment (andriussvilys) [reply css/main.css:9]: Copied css straight out of google calendar without thinking, bad practice obviously

File: css/main.css (hunk 2 above)
Comment (andriussvilys) [reply css/main.css:93]: using BEM convention, modifier should separated with _

File: month/months.js (hunk 7 above)
Comment (andriussvilys) [reply month/months.js:38]: changed: findElemByDate -> findMonthButtonByDate

File: js/weekView.js [hunk 16]
```diff
@@ -1,9 +1,159 @@
...
+    headerCell.classList = "hour header-cell"
+
+    const hourSeparator = document.createElement('div')
+    hourSeparator.classList = "hour-separator"
+
+    headerCell.appendChild(hourSeparator)
+
+    container.appendChild(headerCell)
+
+    for (let index = 1; index < 12; index++) {
```
[the remark this answers is not in the document]
Comment (andriussvilys) [reply js/weekView.js:102]: createHourCell now takes in date param, extracts hour and decides meridiam value

File: js/weekView.js (hunk 9 above)
Comment (andriussvilys) [reply js/weekView.js:145]: done 😎

File: js/dateManipulation.js [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+
+export const incrementDay = (date, incrValue) => {
+    return new Date(date.getFullYear(), date.getMonth(), date.getDate() + incrValue)
+}
+
+export const incrementMonth = (date, incrValue) => {
+    return new Date(date.getFullYear(), date.getMonth() + incrValue, 15)
+}
+
+export const updateCalendarLabels = ( parentElem, date ) => {
```
Comment (dovilel) [inline js/dateManipulation.js:49]: I think this function belongs in header.js.

your dateManipulation.js is strictly algorithms for date stuff. Keep it separate from your DOM managing logic

Comment (dovilel) [reply js/dateManipulation.js:49]: ok, I see the problem. You used it in multiple places.
Generally I'd say you need a single place to update your "current date" and then frmo that place you call separate functions to update labels, where needed. Like state management, sortof

File: js/weekView.js [hunk 18]
```diff
@@ -1,9 +1,153 @@
...
+
+    const hourSeparator = document.createElement('div')
+    hourSeparator.classList = "hour-separator"
+
+    headerCell.appendChild(hourSeparator)
+
+    container.appendChild(headerCell)
+
+    for (let index = 1; index < HOUR_COUNT; index++) {
+        const cell = createHourCell(new Date(0, 0, 0, 0 + index))
```
Comment (dovilel) [inline js/weekView.js:103]: what happens when you add 0 to index? :D

File: js/weekView.js [hunk 19]
```diff
@@ -1,9 +1,153 @@
...
+
+    headerLabel.appendChild(weekday)
+    headerLabel.appendChild(button)
+
+    headerCell.appendChild(headerLabel)
+    headerCell.appendChild(eventCell)
+
+    columnContainer.appendChild(headerCell)
+
+    for (let index = 0; index < ROW_COUNT-1; index++) {
```
Comment (dovilel) [inline js/weekView.js:38]: missed this ROW_COUNT

**Pull request merged**